src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -256,12 +258,21 @@ private function checkClass($class, $file = null)
                     @trigger_error(sprintf('The "%s::%s()" method is considered final%s. It may change without further notice as of its next major version. You should not extend it from "%s".', $declaringClass, $method->name, $message, $name), E_USER_DEPRECATED);
                 }
 
-                foreach ($parentAndTraits as $use) {
-                    if (isset(self::$internalMethods[$use][$method->name])) {
-                        list($declaringClass, $message) = self::$internalMethods[$use][$method->name];
-                        if (\strncmp($ns, $declaringClass, $len)) {
-                            @trigger_error(sprintf('The "%s::%s()" method is considered internal%s. It may change without further notice. You should not extend it from "%s".', $declaringClass, $method->name, $message, $name), E_USER_DEPRECATED);
-                        }
+                if (isset(self::$internalMethods[$name][$method->name])) {
```

Comment:
Unrelated change but the `foreach` was useless since we merge parent + traits infos a few lines above.